### PR TITLE
[AND-2664] Support smart and adaptive banners in Admob adapters

### PR DESCRIPTION
**Pull request by HuFangshuai**
Have been created this PR for https://vungle.atlassian.net/browse/AND-2664
The Admob adapter code needs to be able to support banners of varying heights and widths; and make adjustments based on Vungle supported sizes; instead rejecting sizes that are not Vungle compliant.

### CODE REVIEW

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 1]
```diff
@@ -337,6 +338,14 @@ private void createBanner() {
             vungleBannerAd = mVungleManager.getVungleBanner(mPlacementForPlay, mAdConfig.getAdSize(), mVunglePlayListener);
             if (vungleBannerAd != null) {
                 updateVisibility();
+                RelativeLayout.LayoutParams adViewParams = (RelativeLayout.LayoutParams) vungleBannerAd.getLayoutParams();
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java]: For smart/adaptive banners, `adLayout` width maybe larger than `vungleBannerAd` so we need let subview getting centered. Detailed comments see: https://vungle.atlassian.net/browse/AND-2664

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 2]
```diff
@@ -399,4 +431,61 @@ private boolean hasBannerSizeAd(AdSize adSize) {
         return adSizeType != null;
     }
 
+    // Start of helper code to remove when available in SDK
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java]: See FB's implementation: https://github.com/googleads/googleads-mobile-android-mediation/blob/ebce3b3ccf1c7a0cd8ecb31819c0037b8885d584/ThirdPartyAdapters/facebook/facebook/src/main/java/com/google/ads/mediation/facebook/FacebookAdapter.java#L760

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 3]
```diff
@@ -245,11 +246,6 @@ public void requestBannerAd(Context context,
         }
 
         adLayout = new RelativeLayout(context);
-        // Make adLayout wrapper match the requested ad size, as Vungle's ad uses MATCH_PARENT for
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java]: For smart/adaptive banner/MREC we don't need to let wrapper view match the request ad size. Let's do it on app side.

File: ThirdPartyAdapters/vungle/vungle/build.gradle [hunk 4]
```diff
@@ -36,8 +36,13 @@ android {
 }
 
 dependencies {
-    implementation ('com.vungle:publisher-sdk-android:6.5.1') {
-        transitive=true
+    if (gradle.ext.has('dependOnVungleSdkProject') && gradle.ext.dependOnVungleSdkProject) {
+        implementation project(':vungle-android-sdk:publisher-sdk-android')
+    } else {
+        implementation ('com.github.vungle:vungle-android-sdk:6.5.2-RC3') {
```
Comment (palrahul) [inline ThirdPartyAdapters/vungle/vungle/build.gradle]: can we bump this to RC4 , thats our latest with some fixes for MOAT's new View ability 3rd party lib

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 5]
```diff
@@ -19,6 +19,7 @@
 import com.vungle.warren.VungleBanner;
```
Comment (palrahul) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java]: as we have copied some code as-is we should include a link to the FB code/file and add the license from that file to this Vungle src code file
https://github.com/googleads/googleads-mobile-android-mediation/blob/ebce3b3ccf1c7a0cd8ecb31819c0037b8885d584/ThirdPartyAdapters/facebook/facebook/src/main/java/com/google/ads/mediation/facebook/FacebookAdapter.java#L760

File: ThirdPartyAdapters/vungle/vungle/build.gradle (hunk 4 above)
Comment (HuFangshuai) [reply ThirdPartyAdapters/vungle/vungle/build.gradle]: fixed

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java (hunk 5 above)
Comment (HuFangshuai) [reply ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java]: fixed.

**Pull request merged**